### PR TITLE
Os removal

### PR DESCRIPTION
**Pull request by vigji**
Refactoring that purges the `os` dependency for working with the filesystem in favour of the more modern `pathlib`, as per #299.

Since I was not sure of the temporal horizon of the removal of importing modules I just did the refactoring everywhere. 

I also included a small refactoring to avoid the code duplications in the `save` methods across `Ts`X classes.

### CODE REVIEW

File: tests/test_interval_set.py [hunk 1]
```diff
@@ -511,8 +508,8 @@ def test_save_npz():
     np.testing.assert_array_almost_equal(file['end'], end)
```
[the remark this answers is not in the document]
Comment (BalzaniEdoardo) [reply tests/test_interval_set.py:508]: Also a test that is not touched by this PR (test_interval_set::test_as_units) is failing now, and it is a problem of dtype:

```
FAILED tests\test_interval_set.py::test_as_units - AssertionError: DataFrame.index are different

Attribute "dtype" are different
[left]:  int64
[right]: int32
```

This change should fix that too:

```python
def test_as_units():
    ep = nap.IntervalSet(start=0, end=100)
    df = pd.DataFrame(data=np.array([[0.0, 100.0]]), columns=["start", "end"], dtype=np.float64)
    pd.testing.assert_frame_equal(df, ep.as_units("s").astype(np.float64))
    pd.testing.assert_frame_equal(df * 1e3, ep.as_units("ms").astype(np.float64))
    tmp = df * 1e6
    np.testing.assert_array_almost_equal(tmp.values, ep.as_units("us").values.astype(np.float64))
```

**Review by BalzaniEdoardo — COMMENTED** [review]
I left a couple of fixes for the tests. I am a bit puzzled about why this tests fail only on windows, including a test that you did not edit and used to pass

File: tests/test_interval_set.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vigji) [reply tests/test_interval_set.py:508]: Yeah the test structure could definitively have some proper cleanup using temporary directories and fixtures (apart from linting which is currently missing there)! 

Will include those changes for now

**Comment by vigji** [comment]
maybe we can do it here, or if you prefer in a future PR by you, but I would totally lint tests! Let me know it could be done here.

**Review by gviejo — APPROVED** [review]
Great PR thank you Luigi

**Comment by gviejo** [comment]
Let's keep the linting of the tests for another PR

**Comment by BalzaniEdoardo** [comment]
@vigji I echo Guillaume, thanks for this!

**Pull request merged**